assets/scripts/characters/mario/Mario.ts

```diff
@@ -0,0 +1,378 @@
+// Learn TypeScript:
+//  - https://docs.cocos.com/creator/manual/en/scripting/typescript.html
+// Learn Attribute:
+//  - https://docs.cocos.com/creator/manual/en/scripting/reference/attributes.html
+// Learn life-cycle callbacks:
+//  - https://docs.cocos.com/creator/manual/en/scripting/life-cycle-callbacks.html
+
+import Character from '../Character';
+import DIRECTION, { DIRECTIONS } from '../DIRECTION';
+import IDyingCharacter from '../IDyingCharacter';
+import MARIO_STATES from './MARIO_STATES';
+import MOVEMENT_TYPE from './MOVEMENT_TYPE';
+
+const { ccclass } = cc._decorator;
+
+@ccclass
+export default class Mario extends Character implements IDyingCharacter {
+  private isAlive = true;
+
+  public jumpForce = 9000;
+
+  public moveForce = 600;
+
+  public maxSpeed = 200;
+
+  public walkingMaxSpeed = 200;
+
+  private _runHoldDown = false;
+
+  public get runHoldDown(): boolean {
+    return this._runHoldDown;
+  }
+
+  public set runHoldDown(value: boolean) {
+    if (this.runHoldDown !== value) {
+      this._runHoldDown = value;
+
+      if (!this.runHoldDown) {
+        this.maxSpeed = this.walkingMaxSpeed;
+      } else this.maxSpeed = this.walkingMaxSpeed * 2;
+    }
+  }
+
+  private _isImmortal = false;
+
+  public get isImmortal(): boolean {
+    return this._isImmortal;
+  }
+
+  // for now, this setter is pretty much useless, but it'll be needed later
+  public set isImmortal(value: boolean) {
+    if (value !== this.isImmortal) {
+      this._isImmortal = value;
+    }
+  }
+
+  private _isSuper = false;
```

Comment:
I know at this level you can only get bigger, but, i think i don't like this way to work in the code using a single variable.

Reply:
In that case, do you have any recommendations?